### ISSUE TITLE
Problems with locale

### ISSUE THREAD

**Issue by aNerdgirl** [firefly-iii/firefly-iii#3353]
Dear all,
first of all, thank you very much for all the work. It is an amazing project!
I installed firefly in laragon to try it out.
Unfortunately, I get the error, that the local isn't installed correctly (it is also in the Error FAQ of the documentation, under 'Numbers aren't shown correctly'.
https://docs.firefly-iii.org/faq/self_hosted#decimal-points-are-missing-numbers-are-off

My server has the following local installed (for laragon it is the same as the windows machine):
LCID Name DisplayName
1033 en-US English (United States)

Therefore I went to
firefly-iii\resources\lang\en_US
and change hte config.php:
'locale' => 'en-US',

But the error stays the same. Do I still need to do something? Or do I need to delete the other, non used, languages?
Thank you in advance and sorry if it is an obvious issue.

**Comment by JC5** [comment]
You're very welcome.

I assume that your language is set to English (US) and that you left the locale setting on "(equal to language)". Correct?

You don't need to remove the other languages. But keep in mind that locale settings on Windows are crap and may not be supported properly yet.

**Comment by aNerdgirl** [comment]
Yes, correct.
But the error / blue banner stays and the the decimales aren't shown.

But don't worry. I will probably buy the new rasperry pie 4 and install firefly on it. That should work smoother.

**Comment by skibbipl** [comment]
Sorry for hijacking the thread but I have same issue. I tried to switch to PL locale and numbers are broken. I have all locales set correctly on the system:
```
# locale -a
C
C.UTF-8
en_US.utf8
pl_PL.utf8
POSIX
```

**Comment by dm** [comment]
@JC5 Reviving this thread since found an issue that seems related, using en_GB is showing [this error](https://i.imgur.com/53V4MCr.png) in locale (brand new setup). Triggered [here](https://github.com/firefly-iii/firefly-iii/blob/09990acaa281521a66b3bae66996dd4aca3b16e5/app/Http/Middleware/Range.php#L90).

**Bug description**
Running v5.2.6 through docker (jc5x/firefly-iii:stable, 10b28d21c3e5)
Database is Postgres, setup is ran through [docker-compose.yml](https://gist.github.com/dm/50c43fb536406f37a1555f0ed6f4d29c), a bit cleaned up, it's generated from the following [jinja template](https://gitlab.com/NickBusey/HomelabOS/-/blob/dev/roles/firefly_iii/templates/docker-compose.firefly_iii.yml.j2)

**Steps to reproduce**
Run in docker-compose; in initial setup using British Pound and English (UK). Expected to see number formatting working correctly.

**Extra info**
Running docker on Ubuntu 20.04 LTS, checked locales available in your container image:

```
$ locale -a
C
C.UTF-8
POSIX
cs_CZ.utf8
de_DE.utf8
el_GR.utf8
en_US.utf8
es_ES.utf8
fi_FI.utf8
fr_FR.utf8
hu_HU.utf8
id_ID.utf8
it_IT.utf8
nb_NO.utf8
nl_NL.utf8
pl_PL.utf8
pt_BR.utf8
ro_RO.utf8
ru_RU.utf8
sv_SE.utf8
tr_TR.utf8
vi_VN.utf8
zh_CN.utf8
zh_TW.utf8
```

**Bonus points**
- Since `locale -a` didn't list en_GB; tried installing specific en_GB.UTF-8 locale on the container (dpkg-reconfigure locales). After that was done, the locale is listed:

```
$ locale -a
C
C.UTF-8
POSIX
en_GB.utf8
```

The error does go away (so the locale is loaded), **but the formatting still isn't corrected** ([see result](https://i.imgur.com/KdRcsJD.png))!

- Can run a local debug and investigate further if needed

**Comment by JC5** [comment]
Did you try restarting the container?

**Comment by dm** [comment]
That's a good point, I hadn't restarted the container yet... It does fix it as I would've expected.

Still, would be good to update the container and add en_GB. Shall I add a PR for that?

Cheers

**Comment by JC5** [comment]
Yes, since it’s a language. Check out the base image repository. Thanks!

**Comment by dm** [comment]
I've [done so](https://github.com/firefly-iii/docker-base-image/pull/4), thanks for taking the time. Should also fix #3308 as well.

@JC5 Perhaps I'd argue, James, that since Firefly III is serving multiple people in various locations, it might actually make sense to install `locales-all`; this is the kind of project where that support actually makes sense. 

Cheers 👍

**Comment by JC5** [comment]
I'll think about it. I want to keep the image small but I see the advantages. On the other hand, in the current state the "locale" settings change nothing very interesting for most users. Even the date formatting may not be correct because Firefly III tends to fall back to the very American format of "Sept 12, 2020".

It's on my list nevertheless, thanks for your help!
